lib/modules/platform/gerrit/scm.ts

```diff
@@ -0,0 +1,160 @@
+      );
+    }
+  }
+
+  override async isBranchModified(branchName: string): Promise<boolean> {
+    const filter = this.createFilter('open', branchName);
+    const change = await client
+      .findChanges(filter, true)
+      .then((res) => res.pop());
+    if (change) {
+      const currentGerritPatchset = change.revisions![change.current_revision!];
+      return currentGerritPatchset.uploader.username !== username;
+    }
+    return false;
+  }
+
+  override async commitAndPush(
+    commit: CommitFilesConfig
+  ): Promise<CommitSha | null> {
+    logger.info(`commitAndPush(${commit.branchName})`);
+    const filter = this.createFilter(
+      'open',
+      commit.branchName,
+      commit.baseBranch
+    );
+    const existingChange = await client
+      .findChanges(filter, true)
+      .then((res) => res.pop());
+
+    let hasChanges = true;
+    const origMsg =
+      typeof commit.message === 'string' ? [commit.message] : commit.message;
+    commit.message = [
+      ...origMsg,
+      `Change-Id: ${existingChange?.change_id ?? generateChangeId()}`,
+    ];
+    const commitResult = await git.prepareCommit(commit);
+    if (commitResult) {
+      const { commitSha } = commitResult;
+      if (existingChange?.revisions && existingChange.current_revision) {
+        const fetchRefSpec =
+          existingChange.revisions[existingChange.current_revision].ref;
+        await git.fetchRevSpec(fetchRefSpec); //fetch current ChangeSet for git diff
+        hasChanges = await git.hasDiff('HEAD', 'FETCH_HEAD'); //avoid empty patchsets
+      }
+      if (hasChanges || commit.force) {
+        const pushResult = await git.pushCommit({
+          sourceRef: commit.branchName,
+          targetRef: `refs/for/${commit.baseBranch!}%t=sourceBranch-${
+            commit.branchName
+          }`,
+          files: commit.files,
+        });
+        if (pushResult) {
+          //existingChange was the old change before commit/push. we need to approve again, if it was previously approved from renovate
+          if (
+            existingChange &&
+            client.wasApprovedBy(existingChange, username)
+          ) {
+            await client.approveChange(existingChange._number);
+          }
+          return commitSha;
+        }
+      }
+    }
+    return null; //empty commit, no changes in this Gerrit-Change
+  }
+
+  override deleteBranch(branchName: string): Promise<void> {
+    return Promise.resolve();
+  }
+}
+
+/**
+ * This function should generate a Gerrit Change-ID analogous to the commit hook. We avoid the commit hook cause of security concerns.
+ * random=$( (whoami ; hostname ; date; cat $1 ; echo $RANDOM) | git hash-object --stdin) prefixed with an 'I'.
+ * TODO: Gerrit don't accept longer Change-IDs (sha256), but what happens with this https://git-scm.com/docs/hash-function-transition/ ?
+ */
+function generateChangeId(): string {
+  return 'I' + createHash('sha1').update(randomUUID()).digest('hex');
```

Comment:
use sha256 and trim length instead, otherwise we get to many noise from code analysis.

we already have a util hash function to shorten this too.